src/Text/Pandoc/Writers/Markdown.hs

```diff
@@ -677,6 +669,31 @@ blockToMarkdown' opts (OrderedList (start,sty,delim) items) = do
 blockToMarkdown' opts (DefinitionList items) = do
   contents <- inList $ mapM (definitionListItemToMarkdown opts) items
   return $ mconcat contents <> blankline
+blockToMarkdown' opts (Figure figattr capt body) = do
+  let combinedAttr imgattr = case imgattr of
+        ("", cls, kv) | (figid, [], []) <- figattr -> Just (figid, cls, kv)
+        _ -> Nothing
+  let combinedAlt alt = case capt of
+        Caption Nothing [Plain captInlines]
+          | captInlines == alt -> Just captInlines
+        Caption Nothing [] -> Just alt
+        _ -> Nothing
+  case body of
+    [Plain [Image imgAttr alt (src, ttl)]]
+      | isEnabled Ext_implicit_figures opts
+      , Just descr    <- combinedAlt alt
+      , Just imgAttr' <- combinedAttr imgAttr
+      , isEnabled Ext_link_attributes opts || imgAttr' == nullAttr
+        -> do
+          -- use implicit figures if possible
+          let tgt' = (src, fromMaybe ttl $ T.stripPrefix "fig:" ttl)
+          contents <- inlineListToMarkdown opts [Image imgAttr' descr tgt']
```

Comment:
I think we need a nonempty alt for implicit figures.  So maybe the `_` case for `combinedAlt` should be `Just "image"`?  That preserves earlier behavior.